pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,639 @@
+	}
+
+	for _, exportVolume := range vmexport.Status.Links.External.Volumes {
+		// Access the requested volume
+		if exportVolume.Name == volumeName {
+			for _, format := range exportVolume.Formats {
+				// If the --raw flag is used, we download the uncompressed file
+				if downloadRaw {
+					if format.Format == exportv1.KubeVirtRaw {
+						downloadUrl = format.Url
+						httpHeader = exportTokenHeader
+						break
+					} else if format.Format == exportv1.Dir {
+						downloadUrl = format.Url
+						httpHeader = dirExportTokenHeader
+						break
+					}
+				} else {
+					if format.Format == exportv1.KubeVirtGz || format.Format == exportv1.ArchiveGz {
+						downloadUrl = format.Url
+						httpHeader = exportTokenHeader
+						break
+					}
+				}
+			}
+		}
+	}
+
+	if downloadUrl == "" {
+		return "", "", fmt.Errorf("Unable to get a valid URL from '%s/%s' VirtualMachineExport", vmexport.Namespace, vmexport.Name)
+	}
+
+	return downloadUrl, httpHeader, nil
+}
+
+// waitForVirtualMachineExport waits for the VirtualMachineExport status to be ready
+func waitForVirtualMachineExport(client kubecli.KubevirtClient, namespace string, interval, timeout time.Duration) error {
+	err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		vmexport, err := getVirtualMachineExport(client, namespace)
+		if err != nil {
+			return false, err
+		}
+
+		if vmexport == nil || vmexport.Status == nil {
+			return false, nil
+		}
+
+		done := vmexport.Status.Phase == "Ready"
+		if !done {
+			fmt.Printf("Waiting for VM Export %s status to be ready...\n", vmexportName)
+			return false, nil
+		}
+
+		if vmexport.Status.Links == nil || vmexport.Status.Links.External == nil {
+			fmt.Printf("Waiting for VM Export %s external links to be available...\n", vmexportName)
+			return false, nil
+		}
+
+		fmt.Printf("Processing completed successfully\n")
+		return done, nil
+	})
+
+	return err
+}
+
+// handleHTTPRequest generates the GET request with proper certificate handling
+func handleHTTPRequest(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, downloadUrl, httpHeader string) (*http.Response, error) {
+	token, err := getTokenFromSecret(client, vmexport.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create new certPool and append our external SSL certificate
+	cert := vmexport.Status.Links.External.Cert
+	roots := x509.NewCertPool()
+	roots.AppendCertsFromPEM([]byte(cert))
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{RootCAs: roots},
+	}
+	httpClient := httpClientCreatorFunc(transport)
+
+	// Generate and do the request
+	req, _ := http.NewRequest("GET", downloadUrl, nil)
+	req.Header.Set(httpHeader, token)
+
+	return httpClient.Do(req)
+}
+
+// getHTTPClient assigns the default, non-mocked HTTP client
+func getHTTPClient(transport *http.Transport) *http.Client {
+	if transport == nil {
+		transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
```

Comment:
Please don't set this, it disables host verification, and we provide the cert explicitly so you can verify the host.

Reply:
Since you provide an `--insecure` flag, you can set it if that flag is et.